src/utils/helper_classes/fields/fields_builder.ts

```diff
@@ -0,0 +1,151 @@
+import {AppCall, AppField, AppCallValues} from 'mattermost-redux/types/apps';
+import {AppFieldTypes} from 'mattermost-redux/constants/apps';
+
+export interface Ibuilder {
+    addField(field: AppField): void;
+    addFields(field: AppField[]): void;
+    currentFieldValuesAreDefined(): boolean;
+    getFields(): AppField[];
+    getFieldValueByName(fieldName: string): string;
+    getFieldLabelByName(fieldName: string): string;
+    setDefaultMaxLength(value: number): void;
+    setDefaultMinLength(value: number): void;
+}
+
+export function newFieldsBuilder(call: AppCall): Ibuilder {
```

Comment:
do we need it?

Reply:
I do use these in the code base and feel it was a point of confusion for me early on.  Propose creating a ticket to discuss and address in a separate PR later.

Reply:
@catalintomai, Jira issue created https://mattermost.atlassian.net/browse/MM-33638